compare_test.go

```diff
@@ -592,3 +595,86 @@ func TestOnlyStrict(t *testing.T) {
 		})
 	}
 }
+
+func TestCompareOSFeatures(t *testing.T) {
+	for _, tc := range []struct {
+		platform  string
+		platforms []string
+		expected  []string
+	}{
+		{
+			"linux/amd64",
+			[]string{"windows/amd64", "linux/amd64", "linux(+other)/amd64", "linux/arm64"},
+			[]string{"linux/amd64", "linux(+other)/amd64", "windows/amd64", "linux/arm64"},
+		},
+		{
+			"linux(+none)/amd64",
+			[]string{"windows/amd64", "linux/amd64", "linux/arm64", "linux(+other)/amd64"},
+			[]string{"linux/amd64", "linux(+other)/amd64", "windows/amd64", "linux/arm64"},
+		},
+		{
+			"linux(+other)/amd64",
+			[]string{"windows/amd64", "linux/amd64", "linux/arm64", "linux(+other)/amd64"},
+			[]string{"linux(+other)/amd64", "linux/amd64", "windows/amd64", "linux/arm64"},
```

Comment:
Do we need a case where the `OSVersion` prefixes the `OSFeatures`? ie `linux(7.2+other)`

Reply:
I can add such a case

Reply:
We should define how the OS version can be generically supported as well, possibly similar what is being done with OS Features here